### PR TITLE
Daemon that will screenshot desktops and expose them through sockets

### LINKED ISSUE #3: Daemon that will screenshot desktops and expose them through socket interface
The daemon has to
- Make and store Pixmaps of each virtual desktop together with the _NET_DESKTOP_NAME
- Send Pixmaps to GUI client over unix sockets
  - Make a socket server (daemon) and a socket client (GUI)

### PR DESCRIPTION
**Pull request by mmskv**
Daemon will run in the background and screenshot desktops.
When it receives a request to share screenshots, it will send them over sockets.

### CODE REVIEW

File: src/desktop_pixmap.hpp [hunk 1]
```diff
@@ -34,11 +34,8 @@ class desktop_pixmap : public drawable
   );
   ~desktop_pixmap ();
 
-  // Explicitly deleting unused constructors to comply with rule of five
-  desktop_pixmap (const desktop_pixmap &other) = delete;
-  desktop_pixmap (desktop_pixmap &&other) noexcept = delete;
-  desktop_pixmap &operator= (const desktop_pixmap &other) = delete;
-  desktop_pixmap &operator= (desktop_pixmap &&other) = delete;
+  // Declaring copy operator to use it in vector
+  desktop_pixmap (const desktop_pixmap &);
```
Comment (TheHussky) [inline src/desktop_pixmap.hpp:38]: Are you sure this one covers all the rule of five?

File: src/drawable.hpp [hunk 2]
```diff
@@ -21,13 +21,6 @@ class drawable
             uint16_t width, ///< Width of the display
             uint16_t height ///< Height of the display
   );
-  ~drawable ();
```
Comment (TheHussky) [inline src/drawable.hpp:24]: are  you sure we shouldn't write something instead of deleting everything

File: src/daemon.cpp [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+
+  int prop_length = xcb_get_property_value_length (prop_reply.get ());
+
+  auto *prop = prop_length ? (reinterpret_cast<uint32_t *> (
+                   xcb_get_property_value (prop_reply.get ())))
+                           : throw;
+
+  std::vector<uint32_t> atom_data;
+
+  for (int i = 0; i < prop_length; i++)
```
Comment (TheHussky) [inline src/daemon.cpp:49]: oh, great idea btw

File: src/daemon.cpp [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+struct monitor_info
+{
+  int x;
+  int y;
+  int width;
+  int height;
+};
+
+std::vector<monitor_info>
+get_monitors ()
```
Comment (TheHussky) [inline src/daemon.cpp:74]: I suggest adding some comments/documentation here

File: src/daemon.cpp [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+ * FIXME Uses EWMH and won't work if WM doesn't support large desktops
+ */
+std::vector<desktop_info>
+get_desktops ()
+{
+  auto monitors = get_monitors ();
+
+  auto number_of_desktops = get_property_value ("_NET_NUMBER_OF_DESKTOPS")[0];
+
+  auto viewport = get_property_value ("_NET_DESKTOP_VIEWPORT");
```
Comment (TheHussky) [inline src/daemon.cpp:140]: I don't think this will work on my os

**Review by TheHussky — APPROVED** [review]

File: src/desktop_pixmap.hpp (hunk 1 above)
Comment (mmskv) [reply src/desktop_pixmap.hpp:38]: iirc when you explicitly declare copy constructor, move constructor won't be declared implicitly
same for assignment operator

File: src/drawable.hpp (hunk 2 above)
Comment (mmskv) [reply src/drawable.hpp:24]: sure. It will use just use defaults and defaults fit our case

File: src/daemon.cpp (hunk 5 above)
Comment (mmskv) [reply src/daemon.cpp:140]: Compositing DEs are not really a target demographics here. You already have you desktop previewer built in

File: src/daemon.cpp (hunk 4 above)
Comment (mmskv) [reply src/daemon.cpp:74]: sure

**Pull request merged**